bigquery_to_datadog.py

```diff
@@ -0,0 +1,175 @@
+                    , "indexer" AS server_address
+                FROM `dydx-ce5e3.latency_experiments.long_running_two_sided_orders` s
+                JOIN `dydx-ce5e3.indexer_stream.received_orders_and_cancels` p
+                    ON p.client_id = CAST(s.client_id AS STRING)
+                   AND p.address = s.address
+                   AND p.received_at > TIMESTAMP("{start_timestamp}")
+                  AND p.received_at > TIMESTAMP(@timestamp)
+                WHERE s.sent_at > TIMESTAMP("{start_timestamp}")
+                AND s.address = @maker_address
+                ORDER BY 1
+        """.format(
+            start_timestamp=START_TIMESTAMP
+        ),
+        "params": {"maker_address": config["maker_address"]},
+        "metric_name": "bigquery.short_term_order_latency",
+    },
+    {
+        "name": "stateful_order_latency",
+        "query": """
+            SELECT p.received_at
+                , TIMESTAMP_DIFF(p.received_at, s.sent_at, millisecond) AS latency
+                , p.server_address
+            FROM `dydx-ce5e3.latency_experiments.long_running_stateful_orders` s
+            JOIN `dydx-ce5e3.full_node_stream.order_places` p
+                ON p.client_id = CAST(s.client_id AS STRING)
+            AND p.address = s.address
+            AND p.received_at > TIMESTAMP("{start_timestamp}")
+            AND p.received_at > TIMESTAMP(@timestamp)
+            WHERE s.sent_at > TIMESTAMP("{start_timestamp}")
+            AND s.address = @stateful_address
+            UNION ALL
+            SELECT p.received_at
+                , TIMESTAMP_DIFF(p.received_at, s.sent_at, millisecond) AS latency
+                , "indexer" AS server_address
+            FROM `dydx-ce5e3.latency_experiments.long_running_stateful_orders` s
+            JOIN `dydx-ce5e3.indexer_stream.received_orders_and_cancels` p
+                ON p.client_id = CAST(s.client_id AS STRING)
+            AND p.address = s.address
+            AND p.received_at > TIMESTAMP("{start_timestamp}")
+            AND p.received_at > TIMESTAMP(@timestamp)
+            WHERE s.sent_at > TIMESTAMP("{start_timestamp}")
+            AND s.address = @stateful_address
+            ORDER BY 1
+        """.format(
+            start_timestamp=START_TIMESTAMP
+        ),
+        "params": {"stateful_address": config["stateful_address"]},
+        "metric_name": "bigquery.stateful_order_latency",
+    },
+]
+
+DATADOG_API_KEY = config["dd_api_key"]
+DATADOG_APP_KEY = config["dd_app_key"]
+PROJECT_ID = config["bigquery_project_id"]
+POLL_INTERVAL = 60  # 1 minute
+STATE_FILE = "last_processed_timestamps.json"
+
+# Initialize Datadog API
+options = {"api_key": DATADOG_API_KEY, "app_key": DATADOG_APP_KEY}
+initialize(**options)
+
+
+def load_last_processed_timestamps():
+    if os.path.exists(STATE_FILE):
+        with open(STATE_FILE, "r") as f:
+            return json.load(f)
+    return {query["name"]: START_TIMESTAMP for query in QUERIES}
```

Comment:
possibly not worth the effort, but is there a way to get last written timestamp from dd? that way if the file gets deleted, the query doesn't go back to the start

Reply:
yeah i figured this is the easiest, not a huge deal to just change the hardcoded date if we want to in the future